pkg/controller/operators/olm/requirements.go

```diff
@@ -70,165 +69,65 @@ func (a *Operator) minKubeVersionStatus(name string, minKubeVersion string) (met
-			status.Status = "NotPresent"
-			met = false
-			statuses = append(statuses, status)
-			continue
-		}
-
-		// Check if APIService is registered
-		apiService, err := a.lister.APIRegistrationV1().APIServiceLister().Get(name)
-		if err != nil {
-			status.Status = "NotPresent"
+		g := strings.SplitN(r.Name, ".", 2)
+		if len(g) != 2 {
 			met = false
-			statuses = append(statuses, status)
-			continue
+			err = fmt.Errorf("CRD name %s is not in the form of `pural.group`", r.Name)
+			return
 		}
 
-		// Check if API is available
-		if !a.isAPIServiceAvailable(apiService) {
-			status.Status = "NotPresent"
-			met = false
-		} else {
-			status.Status = "Present"
-			status.UUID = string(apiService.GetUID())
-		}
-		statuses = append(statuses, status)
+		resourceGVKNs = append(resourceGVKNs, v1alpha1.RequirementStatus{
+			Group:   g[1],
+			Version: r.Version,
+			Kind:    r.Kind,
+			Name:    r.Name,
+		})
 	}
 
-	// Check owned API services
-	for _, r := range ownedAPIServiceDescs {
-		name := fmt.Sprintf("%s.%s", r.Version, r.Group)
-		status := v1alpha1.RequirementStatus{
-			Group:   "apiregistration.k8s.io",
-			Version: "v1",
-			Kind:    "APIService",
-			Name:    name,
-		}
-
-		found := false
-		for _, spec := range strategyDetailsDeployment.DeploymentSpecs {
-			if spec.Name == r.DeploymentName {
-				status.Status = "DeploymentFound"
-				statuses = append(statuses, status)
-				found = true
-				break
-			}
-		}
-
-		if !found {
-			status.Status = "DeploymentNotFound"
-			statuses = append(statuses, status)
-			met = false
-		}
+	for _, gvk := range ownedAPIServiceDescs {
+		resourceGVKNs = append(resourceGVKNs, v1alpha1.RequirementStatus{
+			Group:   gvk.Group,
+			Version: gvk.Version,
+			Kind:    gvk.Kind,
+			Name:    fmt.Sprintf("%s.%s", gvk.Version, gvk.Group),
+		})	}
+
+	for _, gvk := range requiredAPIServiceDescs {
+		resourceGVKNs = append(resourceGVKNs, v1alpha1.RequirementStatus{
+			Group:   gvk.Group,
+			Version: gvk.Version,
+			Kind:    gvk.Kind,
+			Name:    fmt.Sprintf("%s.%s", gvk.Version, gvk.Group),
+		})	}
+
+	for _, gvk := range requiredNativeAPIs {
+		resourceGVKNs = append(resourceGVKNs, v1alpha1.RequirementStatus{
+			Group:   gvk.Group,
+			Version: gvk.Version,
+			Kind:    gvk.Kind,
+			Name:    fmt.Sprintf("%s.%s", gvk.Version, gvk.Group),
+		})
 	}
 
-	for _, r := range requiredNativeAPIs {
-		name := fmt.Sprintf("%s.%s", r.Version, r.Group)
-		status := v1alpha1.RequirementStatus{
-			Group:   r.Group,
-			Version: r.Version,
-			Kind:    r.Kind,
-			Name:    name,
-		}
-
-		if err := a.isGVKRegistered(r.Group, r.Version, r.Kind); err != nil {
+	for _, status := range resourceGVKNs {
+		if err := a.isGVKRegistered(status.Group, status.Version, status.Kind); err != nil {
 			status.Status = v1alpha1.RequirementStatusReasonNotPresent
-			status.Message = "Native API does not exist"
+			status.Message = "resource does not exist"
 			met = false
```

Comment:
@njhale I wonder if my understanding is wrong on the count that NativeAPIs are considered optional? It apparently still affects the requirements if the GVK of a nativeAPI is not found. Did I understand something wrong? Thanks.